### PR TITLE
Lesson4

### PR DESCRIPTION
**Pull request by DmitrTRC**
### Lesson3 HomeTasks: Completed

Awaiting for review

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Comment by DmitrTRC** [comment]
Still waiting for the review...  :)

File: Lesson4/Generics/SimpleGenericCache.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            CreationTime = DateTime.Now,
+            Timeout = timeout
+        };
+    }
+
+    internal T? Fetch(string key) // Return default if not found
+    {
+        if (!_cache.TryGetValue(key, out var cachedValue)) return default;
+
+        if (cachedValue != null && cachedValue.CreationTime.AddSeconds(cachedValue.Timeout) > DateTime.Now)
```
Comment (avchirkin) [inline Lesson4/Generics/SimpleGenericCache.cs]: "больше или равно" только

**Review by avchirkin — APPROVED** [review]
Поправить сравнение с текущим временем, остальное ок

File: Lesson4/Generics/SimpleGenericCache.cs (hunk 1 above)
Comment (DmitrTRC) [reply Lesson4/Generics/SimpleGenericCache.cs]: Fixed

**Pull request merged**